src/arrayMethodReduce.js

```diff
@@ -5,7 +5,16 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = arguments.length > 1 ? initialValue : this[0];
+    const firstElement = arguments.length > 1 ? 0 : 1;
+
+    for (let i = firstElement; i < this.length; i++) {
+      if (this[i] !== '') {
+        accumulator = callback(accumulator, this[i], i, this);
+      }
```

Comment:
Why do you need this check? Any value needs to be processed by a callback function.

Reply:
Done:)